structuretoolkit/analyse/symmetry.py

```diff
@@ -357,9 +357,13 @@ def get_primitive_cell(
         new_structure = self._structure.copy()
         new_structure.cell = cell
         new_structure = new_structure[: len(indices)]
-        new_structure = structuretoolkit.common.helper.set_indices(
-            structure=new_structure, indices=indices
-        )
+        indices_dict = {
+            v: k
+            for k, v in structuretoolkit.common.helper.get_species_indices_dict(
+                structure=self._structure
+            ).items()
+        }
+        new_structure.symbols = [indices_dict[i] for i in indices]
```

Comment:
I think this needs to run over all indices, and I don't think it will, and as far as I can tell that's because `get_species_indices_dict` behaves in an undesirable way by overloading keys.

Note:
```python
import structuretoolkit
from ase.build import bulk

structure = bulk("Al", cubic=True)
structure.symbols[0] = "Mg"
structure.symbols[2] = "Cu"
print(structuretoolkit.common.helper.get_species_indices_dict(structure))
>>> {'Mg': 0, 'Al': 1, 'Cu': 2}
# Yikes! atom 3 being Al is totally missed

# Modified from this PR:
indices_dict = {
             v: k
             for k, v in structuretoolkit.common.helper.get_species_indices_dict(
                 structure=structure
             ).items()
         }
print(indices_dict)
>>> {0: 'Mg', 1: 'Al', 2: 'Cu'}
# The third atom doesn't get its species updated
```

Reply:
I have the feeling you confuse species and symbols. The function you are looking for is:
```
structuretoolkit.common.helper.get_structure_indices(structure)
```
Which returns `array([0, 1, 2, 1])`

Reply:
No, not quite. My confusion was that the `indices` variable is `[0, 1, 2, 1]` and not the array indices `[0, 1, 2, 3]` which I had initially assumed. So my concern was unfounded -- there is no mismatch in the span of integers in `indices` and `indices_dict`.

You did, however, point me nicely in the direction of the helper module. Is it not possible to replace these seven lines with [`structuretoolkit.common.helper.set_indices(new_structure, indices)`](https://github.com/pyiron/structuretoolkit/blob/b3e86b6bf4a2f6c569fc57dad36916c9da371e7d/structuretoolkit/common/helper.py#L122)? I can see this is not _exactly_ the same, because in the existing change the `get_species_indices_dict` call gets `self._structure` while the structure getting its symbols set is `new_structure`, and in my proposal it's just `new_structure` both times. However, `new_structure = [: len(indices)]` and `get_species_indices_dict` is pulling the indices of _first occurrence_ for each species, so I don't see how these can differ.